datafusion/proto/gen/src/main.rs

```diff
@@ -28,6 +28,7 @@ fn main() -> Result<(), String> {
     let descriptor_path = proto_dir.join("proto_descriptor.bin");
 
     prost_build::Config::new()
+        .protoc_arg("--experimental_allow_proto3_optional")
```

Comment:
Should be able to remove this after https://github.com/apache/arrow-datafusion/issues/8602 is fixed

Reply:
Removed merged main, and confirmed that 8602 is fixed for me as well